mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -561,6 +551,46 @@ def _validate_model_signature(self, local_model_path):
                 f"{signature_required_explanation}"
             )
 
+    def _download_model_weights_if_not_saved(self, local_model_path):
+        """
+        Transformers models can be saved without the base model weights by setting
+        `save_pretrained=False` when saving or logging the model. Such 'weight-less'
+        model cannot be directly deployed to model serving, so here we download the
+        weights proactively from the HuggingFace hub and save them to the model directory.
+        """
+        model = _load_model(local_model_path)
+        flavor_conf = model.flavors.get("transformers")
+
+        if not flavor_conf:
+            return
+
+        from mlflow.transformers.flavor_config import FlavorKey
+        from mlflow.transformers.model_io import _MODEL_BINARY_FILE_NAME
+
+        if (
+            FlavorKey.MODEL_BINARY in flavor_conf
+            and os.path.exists(os.path.join(local_model_path, _MODEL_BINARY_FILE_NAME))
+            and FlavorKey.MODEL_REVISION not in flavor_conf
+        ):
+            # Model weights are already saved
+            return
+
+        _logger.info(
+            "It appears that the specified Transformers model was saved without the base model "
+            "weights, but such model cannot be directly registered in the Unity Catalog. "
+            "Attempting to download the model weights from the HuggingFace hub and save them to "
+            "the model directory for enabling model registration."
```

Comment:
```suggestion
            "You are attempting to register a transformers model that does not have persisted model weights. Attempting to fetch the weights so that the model can be registered within Unity Catalog.."
```
We can probably be a bit less verbose for this info message :) It will be seen frequently

Reply:
Makes a lot sense, info log should be as concise as possible.